packages/solana/lib/src/rpc/json_rpc_client.dart

```diff
@@ -1,25 +1,29 @@
 import 'dart:async';
 import 'dart:convert';
+import 'dart:io';
 
-import 'package:http/http.dart';
 import 'package:solana/src/exceptions/http_exception.dart';
 import 'package:solana/src/exceptions/json_rpc_exception.dart';
-import 'package:solana/src/exceptions/rpc_timeout_exception.dart';
 import 'package:solana/src/rpc/json_rpc_request.dart';
 
 class JsonRpcClient {
   JsonRpcClient(
     this._url, {
     required Duration timeout,
     required Map<String, String> customHeaders,
+    HttpClient? httpClient, // Optional client for proxy support.
```

Comment:
I forgot about one more thing, let's use `Client` from `http` package, so that it automatically handles web/native use cases.

Reply:
The `package:http` Client only supports HTTP proxies, not SOCKS proxies.  However, today I dealt with a similar problem with the Solana library, which uses an http.Client widely: you can make an http.Client with dart:io's HttpClient which does support SOCKS proxies, but you can't make a SOCKS-supporting io.HttpClient from an http.Client.

Reply:
I can do like this: https://github.com/cypherstack/stellar_flutter_sdk/blob/eca1d730e952cf6a6d64502f977cfc03876b75d4/lib/src/stellar_sdk.dart#L43-L50

but I can't do the reverse.  http.Client is a subset of dart:io.HttpClient aiui

Reply:
Hm, then that's actually a problem because the library should support web as well. Let me think about how to better handle that.

Reply:
What about a conditional import type of thing? I don't have much (any) experience using Dart/Flutter for web but I have seen some conditional import statements regarding some web stuff. Might be a possibility?

Reply:
Yeah, it could be an option to hide it behind conditional imports, but it will most probably require additional work on creating interfaces and wrapping it all together. Unfortunately, I don't have time to look into it right now, so if anyone has possibility to investigate it, that would be great.

Reply:
We may be able to get around to it at some point. If we do, we'll update this PR and  comment here